### PR TITLE
Fix lone comment

### LINKED ISSUE #142: lone comment in empty struct is incorrect formatted
```
type T struct {
	// foo
}
```

Gets formatted as

```
type T struct { // foo
}
```

### PR DESCRIPTION
**Pull request by sagikazarmark**
Fixes #142

### CODE REVIEW

File: format/format.go [hunk 1]
```diff
@@ -557,7 +557,7 @@ func (f *fumpter) applyPre(c *astutil.Cursor) {
 		f.stmts(node.Body)
 
 	case *ast.FieldList:
-		if node.NumFields() == 0 && f.inlineComment(node.Pos()) == nil {
+		if node.NumFields() == 0 && f.inlineComment(node.Pos()) == nil && len(f.commentsBetween(node.Pos(), node.End())) == 0 {
```
Comment (mvdan) [inline format/format.go:560]:
```suggestion
		if node.NumFields() == 0 && len(f.commentsBetween(node.Pos(), node.End())) == 0 {
```

I think the `inlineComment` check is now redundant, as it's a subset of `commentsBetween` given that we check between the start and end of the entire field list.

**Review by mvdan — APPROVED** [review]
Thanks for the fix! Just one suggestion to simplify the patch.

Also, would you mind squashing your commits into a single one? I generally prefer one commit per PR, in this case including both the test and the fix.

**Comment by sagikazarmark** [comment]
Done!

**Comment by mvdan** [comment]
Thank you! I guess using the `good first issue` label is working :)

**Comment by sagikazarmark** [comment]
It certainly is!

In fact, I chose those issue for a Hacktoberfest promotional video where I demonstrate using `git bisect` to find bugs in a codebase I've never seen before because of that.

**Comment by mvdan** [comment]
Oh, neat. Please share if you publish that!

**Comment by sagikazarmark** [comment]
I'll, although it's in Hungarian. I'm trying to promote Hacktoberfest and OSS in Hungary and native language content is easier to sell and consume apparently. I should probably create an English version, too.

**Pull request merged**